### ISSUE TITLE
[Architecture] [M3] Design student model architecture specification

### ISSUE THREAD

**Issue by atfrank** [atfrank/MLFF-Distiller#19]
## Problem Statement

Design a fast, CUDA-optimized student neural network architecture that can be trained via knowledge distillation from Orb-v2 and FeNNol-PMC teacher models. The student must achieve 5-10x speedup while maintaining >95% accuracy.

## Context

- **Milestone**: M3 (Student Architecture) - creating milestone if not exists
- **Priority**: High
- **Estimated Effort**: 3-5 days (research + design + prototyping)
- **Dependencies**: None (can proceed in parallel with M2 completion)
- **Agent**: Architecture Specialist (Lead) + CUDA Engineer (Consult)

## Rationale

While M2 dataset generation completes (Issue #18), we can begin M3 planning. Student architecture design is on the critical path and requires careful consideration:
- Must be significantly smaller than teachers (Orb-v2 ~100M params)
- Must be CUDA-optimizable (custom kernels feasible)
- Must maintain equivariance and physical constraints
- Must support same inputs/outputs as teachers

**Why Now**: Dataset generation takes time. Parallel architecture design maintains project velocity.

## Acceptance Criteria

- [ ] Comprehensive architecture specification document
- [ ] Model sizing analysis (parameter count, memory footprint)
- [ ] Inference speed estimates (theoretical and benchmark-based)
- [ ] CUDA optimization opportunities identified
- [ ] Comparison with teacher architectures (Orb-v2, FeNNol)
- [ ] Training requirements estimated (data, compute, time)
- [ ] Preliminary PyTorch prototype (sketch, not full implementation)
- [ ] Design review with stakeholders

## Deliverables

1. **Architecture Spec**: `docs/STUDENT_ARCHITECTURE_DESIGN.md` (comprehensive)
2. **Model Sizing**: Parameter counts, memory estimates, FLOPs analysis
3. **Prototype Sketch**: `src/mlff_distiller/models/student_arch_draft.py` (architecture only)
4. **Benchmarks**: Theoretical speedup analysis vs teachers
5. **CUDA Opportunities**: List of kernels for optimization (M4 work)
6. **Comparison Table**: Student vs Orb-v2 vs FeNNol

### Functional Requirements

**Inputs (Must Match Teachers):**
- Atomic positions: (N, 3) float tensor
- Atomic species: (N,) int tensor (atomic numbers)
- Cell parameters: (3, 3) float tensor (periodic systems)

**Outputs (Must Match Teachers):**
- Total energy: scalar (eV)
- Atomic forces: (N, 3) tensor (eV/Å)
- Stress tensor: (3, 3) tensor (eV/Å³)

**Physical Constraints:**
- Equivariant to rotations and translations
- Invariant to permutations of same-species atoms
- Extensive properties (energy scales with system size)
- Energy-force consistency (forces = -∇E)

### Performance Requirements

**Speed Targets:**
- 5-10x faster than Orb-v2 inference
- Target: <5 ms per structure (100-200 atoms) on A100 GPU
- Batch inference: >1000 structures/sec

**Accuracy Targets:**
- Energy MAE: <0.05 eV/atom vs teacher
- Force MAE: <0.1 eV/Å vs teacher
- Stress MAE: <0.5 GPa vs teacher
- >95% correlation with teacher predictions

**Model Size Targets:**
- Parameters: 5-20M (vs Orb-v2 ~100M)
- Memory footprint: <500 MB (vs Orb-v2 ~2 GB)
- Disk size: <100 MB (vs Orb-v2 ~500 MB)

## Preliminary Recommendation: Compact GNN

**Specific Architecture (Draft):**

```
Input: positions (N, 3), species (N,), cell (3, 3)

1. Embedding: Atomic number → 128-dim vector

2. Message Passing (3-4 blocks):
   - Neighbor lists (cutoff=5 Å)
   - Edge features: RBF(distance) + angular
   - Message aggregation per node
   - Equivariant updates

3. Optional: Sparse k-NN Attention (k=20)

4. Energy Readout: MLP → per-atom energies → sum

5. Force Prediction: -∇E (autograd)

Output: energy, forces, stresses
```

**Estimates:**
- Parameters: ~5-10M
- Inference: ~5-10 ms/structure
- Speedup: 5-10x vs Orb-v2

## CUDA Optimization Opportunities (M4)

1. Neighbor list computation (cell lists)
2. Message passing kernels (fused ops)
3. RBF computation (reduce bandwidth)
4. Aggregation (warp reductions)
5. Force computation (fused backward)
6. Batch processing optimization

**Estimated CUDA Gains**: 3-5x → 15-50x total speedup

## Definition of Done

- [ ] Architecture specification complete
- [ ] Model sizing analysis complete
- [ ] PyTorch prototype forward pass works
- [ ] Inference benchmarks on synthetic data
- [ ] CUDA optimization plan documented
- [ ] Design review approved
- [ ] Ready for M3 implementation

## Related Issues

- **Parallel with**: #18 (10K dataset)
- **Enables**: M3 implementation
- **Informs**: M4 CUDA optimization
